ConMan.java

```diff
@@ -1,64 +1,47 @@
+import java.io.InputStream;
 import java.io.PrintWriter;
 import java.io.Writer;
 import java.util.ArrayList;
 
 
 public class ConMan {
     private final UI ui;
+    private final InputStream stream;
     private ConsoleMenu menu;
-    public ArrayList<Contact> allContacts = new ArrayList<>();
+    private ArrayList<Contact> allContacts = new ArrayList<>();
 
-    public ConMan(ConsoleMenu consoleMenu, UI ui) {
+    public ConMan(ConsoleMenu consoleMenu, UI ui, InputStream stream) {
         this.menu = consoleMenu;
         this.ui = ui;
-        //allContacts.add(new Contact("Will", "Curry", "Tele", "Email"));
+        this.stream = stream;
     }
 
     public ArrayList<Contact> allContacts() {
         return allContacts;
     }
 
-    public void shouldContinue() {
-        ui.userContinue();
-        if (ui.userInput().equals("ok")) {
-            start();
-        }
-    }
-
-    private void start() {
+    public void start() {
         Writer writer = new PrintWriter(System.out);
-        menu = new ConsoleMenu(menuItems(), System.in, writer);
+        menu = new ConsoleMenu(commands(), stream, writer);
```

Comment:
Why are you creating a new ConsoleMenu object here when you've passed one in already. 
Maybe this is another reason to just say 
`menu.setCommands(commands)`